src/lib/components/Widget.js

```diff
@@ -0,0 +1,262 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faClock } from "@fortawesome/free-solid-svg-icons";
+import { formatDistanceStrict } from "date-fns";
+import Logo from "./assets/SSWLogo.png";
+import "./styles/style.css";
+
+class Widget extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      error: null,
+      isLoaded: false,
+      rules: [],
+      isDarkMode: props.isDarkMode,
+      author: props.author,
+      numberOfRules: props.numberOfRules ? props.numberOfRules : 10,
+    };
+  }
+
+  capitalizeFirstLetter(string) {
+    return string.charAt(0).toUpperCase() + string.slice(1);
+  }
+
+  extractFromRuleContent(term, text) {
+    var start = text.substring(text.search(`${term}:`));
+    var value = start.substring(term.length + 1, start.search("\\n"));
+    return value.trim();
+  }
+
+  async fetchPullRequests() {
+    const pullRequests = await fetch("https://api.github.com/graphql", {
+      method: "POST",
+      headers: {
+        Authorization: "bearer /* INSERT TOKEN */",
+      },
+      body: JSON.stringify({
+        query: `{
+        rateLimit {
+          remaining
+        }
+        search(query: "repo:SSWConsulting/SSW.Rules.Content is:pr base:main is:merged sort:updated-desc ${
+          this.state.author ? "author:" + this.state.author : ""
+        }", type: ISSUE, first: ${this.state.numberOfRules + 10}) {
+          nodes {
+            ... on PullRequest {
+              author {
+                login
+              }
+              files(first: 10) {
+                nodes {
+                  path
+                }
+              }
+              mergedAt
+            }
+          }
+        }
+      }`,
+      }),
+    }).then((res) => res.json());
+    pullRequests.data.search.nodes.sort((a,b) => new Date(b.mergedAt) - new Date(a.mergedAt))
+    return pullRequests;
+  }
+
+  async fetchFileContents(list) {
```

Comment:
This logic is a bit hard to read, maybe we could break it up a bit more.

Also should the backend logic be done in a separate file?